### PR TITLE
[Java] 4.0 Update Guide

### PR DESCRIPTION
**Pull request by MatKuhr**
- [x] Adds an update guide for v4
- [x] Update the release notes page
- [ ] Link to 4.0 announcement post
- [ ] Actual release notes filled
- [x] Adjust to the new docs versioning

### CODE REVIEW

File: docs-java/guides/4.0-upgrade.mdx [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+  * `com.sap.xs2.security:java-container-security`
+  * `com.sap.conn.jco:com.sap.conn.jco.sapjco3`
+  * `com.sap.conn.jco:sapjco3`
+  * `com.sap.hcp.cf.logging:cf-java-logging-support-logback`
+  * `com.squareup.okhttp3:okhttp`
+
+</details>
+
+
+**If you are using the SAP Java Buildpack (SJB)** together with a `.war` based deployment you should check out our new BOM dedicated to the SJB.
```
[the remark this answers is not in the document]
Comment (MatKuhr) [reply docs-java/guides/4.0-upgrade.mdx:185]: Also thought about this, but when experimenting with the format I found that making it bold felt slightly better for visibility / readability. Maybe something to revisit laster

**Review by KavithaSiva — COMMENTED** [review]
In general, very well written. Drives the point well across that it isn't difficult to upgrade.

File: docs-java/guides/4.0-upgrade.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Johannes-Schneider) [reply docs-java/guides/4.0-upgrade.mdx:184]: I think what's nice about a dedicated chapter/section is the fact that Docusaurus will create an entry in the Table of contents. So customers can jump to that section quickly.

[the remark this answers is not in the document]
Comment (MatKuhr) [reply docs-java/guides/4.0-upgrade.mdx]: turned it into a small section now :)

File: docs-java/guides/4.0-upgrade.mdx [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+
+
+There is a good chance that simply bumping the version to `4.0.0` is already enough, without the need for any further adjustments.
+In particular, updating may be straightforward if you are not using any deprecated API or spawn new threads.
+
+## Before You Update
+
+Before you increase the SAP Cloud SDK version to 4 it can be beneficial to migrate any usage of **deprecated API** first.
+Follow the deprecation note of the Javadoc in case you are still using deprecated API from the SAP Cloud SDK.
+Almost all deprecated API is removed With version 4 (exception being audit logging).
```
[the remark this answers is not in the document]
Comment (MatKuhr) [reply docs-java/guides/4.0-upgrade.mdx]: Done ✅

File: docs-java/guides/4.0-upgrade.mdx [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+    * `ScpCfCloudPlatform#getServiceBindingsRootLocation`
+    * `ScpCfCloudPlatform#setServiceBindingLoader`
+
+</details>
+
+### ThreadContext
+
+:::caution
+The `ThreadContext` interface and all related API have received substantial changes.
+<!-- vale off -->
```
Comment (MatKuhr) [inline docs-java/guides/4.0-upgrade.mdx:498]: Value complains about wordiness here but I disagree. I felt like shortening (e.g. using "often") made the text too blunt, giving the reader not enough "time" to understand it.

**Comment by MatKuhr** [comment]
One open question is what we want to put on the release notes page. We could just put the content from all `details` blocks there. Or we leave it empty and just start with 4.1.0. Or we just link to the update guide. Not sure yet what the best play is...

File: docs-java_versioned_sidebars/version-v3-sidebars.json [hunk 4]
```diff
@@ -73,15 +73,16 @@
       "type": "category",
       "label": "Guides",
       "items": [
-        "guides/cf-cli",
-        "guides/cf-deploy",
-        "guides/cloud-foundry-xsuaa-service",
+        "guides/4.0-upgrade",
```
Comment (MatKuhr) [inline docs-java_versioned_sidebars/version-v3-sidebars.json:76]: I found it to be important that this guide is very visible, so I put it to the top

**Review by Johannes-Schneider — APPROVED** [review]
LGTM

**Pull request merged**